classes/message-handler.js

```diff
@@ -0,0 +1,113 @@
+import EventEmitter from 'events';
+
+/**
+ * Class for handling messages to and from parent frame
+ * 
+ * @param   {Array} origins - List of whitelisted origin domains
+ */
+export default class MessageHandler extends EventEmitter{
+    constructor(origins=['null']) {
+        super();
+        this.origins = origins;
+
+        this.onMessage = this.onMessage.bind(this);
+
+        // Attach listener to window
+        window.addEventListener('message', this.onMessage, false);
+    }
+
+    /** Callback for handling messages */
+    onMessage(event) {
+        // Origin of message event
+        let origin = event.origin;
```

Comment:
Sometimes chrome doesn't use the right origin field btw: https://github.com/matrix-org/matrix-react-sdk/blob/19a7297a770ab4bbe77dda94c4eb3b1c151f50ee/src/FromWidgetPostMessageApi.js#L138-L140

Reply:
Applied the fix. Merging now. :slightly_smiling_face: